### PR TITLE
[AArch64] Implement NEON vscale intrinsics

### PR DESCRIPTION
**Pull request by Lukacma**
This patch implements following intrinsics:

```
float16x4_t vscale_f16(float16x4_t vn, int16x4_t vm)	
float16x8_t vscaleq_f16(float16x8_t vn, int16x8_t vm)
float32x2_t vscale_f32(float32x2_t vn, int32x2_t vm)
float32x4_t vscaleq_f32(float32x4_t vn, int32x4_t vm)
float64x2_t vscaleq_f64(float64x2_t vn, int64x2_t vm)
```

as defined in https://github.com/ARM-software/acle/pull/323

Co-authored-by: Hassnaa Hamdi <hassnaa.hamdi@arm.com>

### CODE REVIEW

File: llvm/lib/Target/AArch64/AArch64InstrFormats.td [hunk 1]
```diff
@@ -5985,6 +5985,27 @@ multiclass SIMDThreeSameVectorFP<bit U, bit S, bits<3> opc,
         [(set (v2f64 V128:$Rd), (OpNode (v2f64 V128:$Rn), (v2f64 V128:$Rm)))]>;
 }
 
+// As above, but only floating point elements supported.
```
[the remark this answers is not in the document]
Comment (SpencerAbson) [reply llvm/lib/Target/AArch64/AArch64InstrFormats.td:5987]: I think the wrong comment was removed here.

File: llvm/lib/Target/AArch64/AArch64InstrFormats.td [hunk 2]
```diff
@@ -5985,6 +5984,27 @@ multiclass SIMDThreeSameVectorFP<bit U, bit S, bits<3> opc,
...
+let mayRaiseFPException = 1, Uses = [FPCR] in
+multiclass SIMDThreeVectorFP<bit U, bit S, bits<3> opc,
```
Comment (SpencerAbson) [inline llvm/lib/Target/AArch64/AArch64InstrFormats.td:5989]: Should the name of this class be changed to reflect that it's second operand's elements must be integers?

Comment (Lukacma) [reply llvm/lib/Target/AArch64/AArch64InstrFormats.td]: Renamed and moved.

File: llvm/lib/Target/AArch64/AArch64InstrFormats.td (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lukacma) [reply llvm/lib/Target/AArch64/AArch64InstrFormats.td]: Reverted

File: clang/include/clang/Basic/arm_neon.td [hunk 3]
```diff
@@ -2096,3 +2096,9 @@ let ArchGuard = "defined(__aarch64__) || defined(__arm64ec__)", TargetGuard = "r
   def VLDAP1_LANE : WInst<"vldap1_lane", ".(c*!).I", "QUlQlUlldQdPlQPl">;
   def VSTL1_LANE  : WInst<"vstl1_lane", "v*(.!)I", "QUlQlUlldQdPlQPl">;
 }
+
+let ArchGuard = "defined(__aarch64__)", TargetGuard = "fp8" in {
```
[the remark this answers is not in the document]
Comment (Lukacma) [reply clang/include/clang/Basic/arm_neon.td]: fixed

**Review by SpencerAbson — APPROVED** [review]
Thanks, LGTM.

**Pull request merged**